### PR TITLE
Add description about the server authentication in the k8s SD

### PR DESCRIPTION
**Pull request by aureamunoz**
…ce discovery strategy

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]

File: docs/kubernetes.md [hunk 1]
```diff
@@ -31,10 +35,9 @@ Stork looks for the _Kubernetes Service_ with the given name (`my-service` in th
...
 
-
 Supported attributes are the following:
 
 | Attribute              | Mandatory    | Default Value     | Description                                                         |
 |-- -------------------- | ------------ | ----------------- | ------------------------------------------------------------------- |
 | `k8s-host`             | No           | _master url_      | The Kubernetes API host                                             |
-| `k8s-namespace`        | No           |                   | The namespace of the service. Use `all` to discover all namespaces. |
\ No newline at end of file
+| `k8s-namespace`        | Yes          |                   | The namespace of the service. Use `all` to discover all namespaces. |
```
Comment (michalszynkiewicz) [inline docs/kubernetes.md:43]: right now if not specified, it defaults to the namespace that is in the .kube/config or the mounted volume, right?

Comment (aureamunoz) [reply docs/kubernetes.md:43]: Yes

Comment (michalszynkiewicz) [reply docs/kubernetes.md:43]: so adding it in the properties is not required

Comment (aureamunoz) [reply docs/kubernetes.md:43]: I should have elaborate more. After [this modification](https://github.com/smallrye/smallrye-stork/blob/30df8133581fee56b8b5282dfb89a2424c686379/service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java#L52), we need to have it in the properties.

Comment (cescoffier) [reply docs/kubernetes.md:43]: still, that does not make it mandatory.

Comment (michalszynkiewicz) [reply docs/kubernetes.md:43]: @aureamunoz it's not needed to have it in properties when it is configured in the environment, then we get it from https://github.com/smallrye/smallrye-stork/blob/30df8133581fee56b8b5282dfb89a2424c686379/service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java#L49

**Pull request merged**